index.html

```diff
@@ -258,6 +295,81 @@ <h2>
+              <p>
+                Otherwise, set |button|.{{GamepadButton/pressed}} to `true` if the value is above the <a>button press threshold</a> or `false` if it is not above the threshold.
+              </p>
+            <li>If the button is capable of detecting touch, set |button|.{{GamepadButton/touched}} to `true` if the button is currently being touched.
+              <p>
+                Otherwise, set |button|.{{GamepadButton/touched}} to |button|.{{GamepadButton/pressed}}.
+              </p>
+            <li>Increment |rawButtonIndex|.
+          </ol>
+        <li>If |navigator|.{{[[hasGamepadGesture]]}} is `false` and |gamepad| <a>contains a gamepad user gesture</a>:
+          <ol>
+            <li>Set |navigator|.{{[[hasGamepadGesture]]}} to `true`.
+            <li>For each |connectedGamepad:Gamepad| of |navigator|.{{[[gamepads]]}}:
+              <ol>
+                <li>If |connectedGamepad| is not `null`:
+                  <ol>
+                    <li>Set |connectedGamepad|.{{Gamepad/timestamp}} to |now| − |gamepadsStart|.
+                    <li>[[=Fire an event=]] named `"gamepadconnected"` at |window:Window| using {{GamepadEvent}} with its {{GamepadEvent/gamepad}} attribute initialized to |connectedGamepad|.
+                  </ol>
+              </ol>
+          </ol>
+      </ol>
+      <p>
+        The <a>user agent</a> MUST provide methods for retrieving <dfn data-lt="ordered array of axis inputs">ordered arrays of axis inputs</dfn> and <dfn data-lt="ordered array of button inputs">button inputs</dfn> for a gamepad such that there is a unique element in the axis array for each axis input and a unique element in the button array for each button input.
+        If the system enumerates axis and button inputs in a consistent order, then the methods SHOULD provide the inputs in that order.
+        Otherwise, the methods MAY use any consistent ordering.
+      </p>
+      <p>
+        The <a>user agent</a> MUST provide methods for retrieving <dfn data-lt="ordered array of minimum logical values">ordered arrays of minimum logical values</dfn> and <dfn data-lt="ordered array of maximum logical values">maximum logical values</dfn> for each gamepad input.
+        The ordering of these arrays MUST match the ordering of the <a>ordered array of axis inputs</a> and <a>ordered array of button inputs</a>.
+      </p>
+      <p>
+        The <a>user agent</a> MUST provide methods for retrieving <dfn data-lt="ordered array of logical axis input values">ordered arrays of logical axis input values</dfn> and <dfn data-lt="ordered array of logical button input values">button input values</dfn>.
+        These arrays MUST contain the most recent values received from the gamepad for each button and axis input.
+        The ordering of these arrays MUST match the ordering of the <a>ordered array of axis inputs</a> and <a>ordered array of button inputs</a>.
+      </p>
```

Comment:
I'd suggest we delete these and just use Infra types. Also, it will avoid descriptors for things that happen algorithmically anyway.

Reply:
Just a note, only add a `data-lt=` to `dfn`s if you are creating an alias.

Like 
```
<dfn data-lt="fruit of the gods">bananas</dfn>  where the [=fruit of the gods=], and yet...
```

Reply:
I've removed these definitions and rewritten the relevant part of the algorithm, please take a look